sonar-update-center-common/src/test/java/org/sonar/updatecenter/common/VersionTest.java

```diff
@@ -72,14 +72,49 @@ public void test_fields_fromString() {
   @Test
   public void compare_releases() {
     Version version12 = Version.create("1.2");
+    Version version13 = Version.create("1.3");
     Version version121 = Version.create("1.2.1");
+    Version version12star = Version.create("1.2.*");
+    Version version2star = Version.create("2.*");
 
     assertThat(version12.toString()).isEqualTo("1.2");
+
+    // Should be compatible with itself.
     assertThat(version12.compareTo(version12)).isEqualTo(0);
+    assertThat(version12.isCompatibleWith(version12)).isTrue();
     assertThat(version121.compareTo(version121)).isEqualTo(0);
-
-    assertThat(version121.compareTo(version12) > 0).isTrue();
-    assertThat(version12.compareTo(version121) < 0).isTrue();
+    assertThat(version12star.compareTo(version12star)).isEqualTo(0);
+    assertThat(version2star.compareTo(version2star)).isEqualTo(0);
+
+    // Should not be compatible with more recent versions.
+    assertThat(version12.compareTo(version13)).isLessThan(0);
+    assertThat(version12.isCompatibleWith(version13)).isFalse();
+    assertThat(version12.compareTo(version121)).isLessThan(0);
+    assertThat(version12.isCompatibleWith(version121)).isFalse();
+
+    // Should not be compatible with less recent versions.
+    assertThat(version121.compareTo(version12)).isGreaterThan(0);
+    assertThat(version121.isCompatibleWith(version12)).isFalse();
+    assertThat(version13.compareTo(version12)).isGreaterThan(0);
+    assertThat(version13.isCompatibleWith(version121)).isFalse();
+
+    // Should be compatible with wildcard versions of the same major release.
```

Comment:
Tiny detail: the comment says "same major" (what is true), while they are same minor.